Change library extension to .dylib
Create library with -dynamiclib
Use -install_name
Use -arch flags
--- src/libcolorer/makefile.gcc.orig	2006-11-24 17:14:40.000000000 -0600
+++ src/libcolorer/makefile.gcc	2014-09-18 20:48:15.000000000 -0500
@@ -1,5 +1,5 @@
 
-TARGET=libcolorer.so
+TARGET=libcolorer.dylib
 all: $(TARGET)
 
 shared_path=../shared/
@@ -8,10 +8,10 @@
 include ../shared/makefile.colorer.gcc
 
 LIBINC = -lstdc++
-LIBFLAGS = $(coreobjects) -shared -o $(TARGET) $(LIBINC)
+LIBFLAGS = $(coreobjects) -dynamiclib -install_name $(prefix)/lib/$(TARGET) -o $(TARGET) $(LIBINC) $(ARCHFLAGS)
 
 
-libcolorer.so: $(coreobjects)
+libcolorer.dylib: $(coreobjects)
 	$(CPP) $(LIBFLAGS)
 
 clean:
